### ISSUE TITLE
Unable to hire additional citizens - not enough homes, when there is

### ISSUE THREAD

**Issue by smmmadden** [ldtteam/minecolonies#6883]
### Prerequisites 

- [x ] I am running the latest **alpha** version of MineColonies and Structurize for my Minecraft version.
- [ x] I checked the MineColonies/Structurize [wiki](https://wiki.minecolonies.ldtteam.com/) and made sure my issue is not covered there.
- [ x] I made sure that **this issue is not a duplicate of any existing issue.**

### Context 

- Minecraft Version: 36.1.4
- MineColonies Version: 0.14.92-ALPHA
- Structurize Version: 0.13.163-ALPHA
- Related Mods and their Versions:

## Expected behavior

If there are spaces for Guard Towers, Barracks then citizens should always be able to join.  The defaults are set to 150 citizens per colony, but my Town Hall (level 5), shows 25/25 citizens which is of course incorrect. 
## Actual behavior

I get a message saying you don't have enough rooms/space for more citizens.

![image](https://user-images.githubusercontent.com/10309346/113964087-d10e7000-97f8-11eb-9ac1-932e3c5c9734.png)
![image](https://user-images.githubusercontent.com/10309346/113964208-074bef80-97f9-11eb-8340-97a1c7f54f66.png)
Separate issue, but the town hall list of jobs would be more helpful if sorted properly.
## Steps to reproduce the issue

1. 
2. 
3. 

### Logs
Latest Log: https://pastebin.com/tgznd048

- latest.log: 
- crashlog: 

### Notes

---
## Viewers

* Add a thumbs-up to the bug report if you are also affected. This helps the bug report become more visible to the team and doesn't clutter the comments.
* Add a comment if you have any insights or background information that isn't already part of the conversation.

**Comment by ravenbuilder934** [comment]
Complete the researches in the university

**Comment by smmmadden** [comment]
No, that does not resolve the issue.   I wish it were that simple. :-)  There is no research to be had in the university as it is stuck on **Level 1** and the few there were, were already done.  This is a builder issue with the University and probably only this style though I'd have to get another colony setup to prove it and we're not done with this one yet.  Most structures cannot be built b/c of the university.  If the dependency is to level up the university before we can add colonies, thats a major change since we didn't see this in 9 other colonies.  This issue is blocked by the other bug reported on the university not able to complete level 2.

**Comment by ravenbuilder934** [comment]
so you didn't complete the researches because you couldn't level up the university? that's a builder issue then, population is working as intended (gated by researches).

**Comment by smmmadden** [comment]
I completed research for Level 1 - but due to not being able to get level 2, everything else is blocked.  Can't add workers even to the Barracks with all level 5 towers and only one guard in each.  Seems odd to restrict population solely on research in the university.  If there is living space, it shouldn't matter IMHO.

**Comment by ravenbuilder934** [comment]
it's based on living space and uni researches

**Comment by smmmadden** [comment]
ok - seems the logic is requiring University over living space.  Meaning that unless the university has reached a certain level, you cannot add more than 25 citizens regardless how many living places there are.  We have 3 houses with no one in them or to put in them and positions for 16 guards in the barracks not counting all the other buildings we're trying to hire for but not able to.

**Comment by ravenbuilder934** [comment]
Yeah that's intentional. I meant if you've completed the researches but don't have enough living space, you can't get more.

**Comment by starfall18** [comment]
Problem, i have level 1 university, room in the barracks and i cant hire additional citizens cause all the houses are full... however if there is one free worker (unemployed) they get employed in the guard towers, releasing space for new citizen, so this does seem like an issue with barracks spaces not being recognized as available living spaces untill someone is assigned to live and work there. For instance, i built a mine that reached my workers cap along with housing space. Before the mine had a worker, i could hire new citizens and they would go and work in the guard towers... But once the mine was finished and hired a worker, the cap was reached and i cant hire more even if there is room for guards. Tested to see if i didnt hit any other cap by firing the mine worker. The mine worker was then assigned to the guard tower and i was again, able to hire more citizens. 
SO WHY??? WHY do i have to fire a citizen from his job to free up a housing space in houses/taverns to hire the guards before i can hire back the the citizen to do his job where he was assigned to?
This bug is still present and hasnt been addressed properly!
